fastapi_template/template/{{cookiecutter.project_name}}/{{cookiecutter.project_name}}/web/api/dummy/views.py

```diff
@@ -14,28 +16,44 @@
 async def get_dummy_models(
     limit: int = 10,
     offset: int = 0,
+{%- if cookiecutter.db_info.name != "mongodb" %}
     dummy_dao: DummyDAO = Depends(),
+{%- endif %}
 ) -> List[DummyModel]:
     """
     Retrieve all dummy objects from the database.
 
     :param limit: limit of dummy objects, defaults to 10.
     :param offset: offset of dummy objects, defaults to 0.
+{%- if cookiecutter.db_info.name != "mongodb" %}
     :param dummy_dao: DAO for dummy models.
+{%- endif %}
     :return: list of dummy objects from database.
     """
+{%- if cookiecutter.db_info.name != "mongodb" %}
     return await dummy_dao.get_all_dummies(limit=limit, offset=offset)
+{%- else %}
+    return await DummyModel.find_all(skip=offset, limit=limit).to_list()
+{%- endif %}
 
 
 @router.put("/")
 async def create_dummy_model(
     new_dummy_object: DummyModelInputDTO,
+{%- if cookiecutter.db_info.name != "mongodb" %}
     dummy_dao: DummyDAO = Depends(),
+{%- endif %}
 ) -> None:
     """
     Creates dummy model in the database.
 
     :param new_dummy_object: new dummy model item.
+{%- if cookiecutter.db_info.name != "mongodb" %}
     :param dummy_dao: DAO for dummy models.
+{%- endif %}
     """
+{%- if cookiecutter.db_info.name != "mongodb" %}
     await dummy_dao.create_dummy_model(name=new_dummy_object.name)
+{%- else %}
+    await DummyModel.insert_one(DummyModel(name=new_dummy_object.name))
```

Comment:
Can we move this into DAO? All databases follow this pattern. Is it a good idea for mongo as well?

Reply:
sure.